fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/FoldConstantRule.java

```diff
@@ -0,0 +1,440 @@
+        Expression child = rewrite(isNull.child(), context);
+        if (child instanceof NullLiteral) {
+            return Literal.of(true);
+        } else if (child instanceof Literal) {
+            return Literal.of(false);
+        }
+        return isNull.withChildren(child);
+    }
+
+    @Override
+    public Expression visitTimestampArithmetic(TimestampArithmetic arithmetic, ExpressionRewriteContext context) {
+        Expression left = rewrite(arithmetic.child(0), context);
+        Expression right = rewrite(arithmetic.child(1), context);
+        if (isAllLiteral(left, right)) {
+            return ExpressionEvaluator.INSTANCE.eval(arithmetic.withChildren(left, right));
+        }
+        return arithmetic.withChildren(left, right);
+    }
+
+    private Expression foldByBe(Expression root, ExpressionRewriteContext context) {
+        if (root.isConstant() && !root.isLiteral()) {
+            Expr expr = ExpressionTranslator.INSTANCE.translate(root, null);
+            assignId(expr);
+            Map<String, Expr> ori = new HashMap<>();
+            ori.put(expr.getId().toString(), expr);
+
+            Map<String, Map<String, TExpr>> paramMap = new HashMap<>();
+            Map<String, TExpr> constMap = new HashMap<>();
+
+            collectConst(expr, constMap);
+            paramMap.put("0", constMap);
+            Map<String, Map<String, Expr>> resMap = calc(paramMap, ori, context.connectContext);
+            Expr result = resMap.get("0").get(expr.getId().toString());
+            if (result instanceof LiteralExpr) {
+                DataType type = DataType.convertFromString(result.getType().getPrimitiveType().toString());
+                return Literal.of(result.getStringValue()).castTo(type);
+            }
+        }
+        return root;
+    }
+
+    private void assignId(Expr expr) {
+        if (expr.getId() == null) {
+            expr.setId(idGenerator.getNextId());
+        }
+        for (Expr child : expr.getChildren()) {
+            assignId(child);
+        }
+    }
+
+    private void collectConst(Expr expr, Map<String, TExpr> constExprMap) {
+        if (expr.isConstant()) {
+            if (expr instanceof CastExpr) {
+                CastExpr castExpr = (CastExpr) expr;
+                if (castExpr.getChild(0) instanceof org.apache.doris.analysis.NullLiteral) {
+                    return;
+                }
+            }
+            if (expr instanceof LiteralExpr) {
+                return;
+            }
+            if (expr instanceof BetweenPredicate) {
+                return;
+            }
+            constExprMap.put(expr.getId().toString(), expr.treeToThrift());
+        }
+        for (int i = 0; i < expr.getChildren().size(); i++) {
+            final Expr child = expr.getChildren().get(i);
+            collectConst(child, constExprMap);
+        }
+    }
+
+    private Map<String, Map<String, Expr>> calc(Map<String, Map<String, TExpr>> map, Map<String, Expr> allConstMap,
```

Comment:
rename to meaningful name like executeExpressionInBackend, and rename the parameter names